### PR TITLE
Simplify test for issue with GHC.Tuple.Unit

### PR DESCRIPTION
**Pull request by hurryabit**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/2475)

### CODE REVIEW

File: compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs [hunk 1]
```diff
@@ -1206,7 +1206,9 @@ ctorLabels flv con =
     | flv `elem` [ClassFlavour, TupleFlavour Boxed] || isTupleDataCon con
       -- NOTE(MH): The line below is a workaround for ghc issue
       -- https://github.com/ghc/ghc/blob/ae4f1033cfe131fca9416e2993bda081e1f8c152/compiler/types/TyCon.hs#L2030
-      || (getOccFS con == "Unit" && GHC.moduleNameFS (GHC.moduleName (nameModule (getName con))) == "GHC.Tuple")
+      -- If we omit this workaround, `GHC.Tuple.Unit` gets translated into a
+      -- variant rather than a record and the `SugarUnit` test will fail.
+      || (getOccFS con == "Unit" && nameModule (getName con) == gHC_TUPLE)
```
Comment (hurryabit) [inline compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs]: @cocreature Unfortunately, comparing with `tupleDataCon Boxed 1` fails although that is exactly what we're looking for. Somehow the uniques don't match. Do you have an _immediate_ idea what could be going wrong?

Comment (cocreature) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs]: @hurryabit Which part exactly is the one that ends up getting a different unique?

Comment (hurryabit) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs]: @cocreature The unique of `con` is different from the unique of `tupleDataCon Boxed 1` even when `con` matches the filter in line 1211. Does that answer your question?

Comment (cocreature) [reply compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs]: Not sure what is going wrong here. Looking at the code, it seems like the unique in `tupleDataCon Boxed 1` should always be fixed so I guess the one in `con` is the one that varies? No idea why :confused:

**Pull request merged**